File: LeetCode/2337/solution.py
# 2337. Move Pieces to Obtain a String
class Solution:
    def canChange(self, start: str, target: str) -> bool:
        n = len(start)
        i, j = 0, 0
    
        while i < n or j < n:
            while i < n and start[i] == '_':
                i += 1
            while j < n and target[j] == '_':
                j += 1

            if i==n or j==n:
                return i==j

            if start[i] != target[j]:
                return False

            if start[i]=='L' and i<j:
                return False
            if start[i]=='R' and i>j:
                return False

            i += 1
            j += 1

        return True

File: LeetCode/2337/solution.cpp
// 2337. Move Pieces to Obtain a String
class Solution {
public:
    bool canChange(string start, string target) {
        int len = start.size();
        for(int i=0, j=0; i<len || j<len; i++, j++){
            while(i<len && start[i]=='_') i++;
            while(j<len && target[j]=='_') j++;

            if(start[i] != target[j]) return false;

            if(start[i]=='L' && i<j) return false;
            if(start[i]=='R' && i>j) return false;
        }
        return true;
    }
};